src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -1123,7 +1127,7 @@ fn lit_to_const<'a, 'tcx>(lit: &'tcx ast::LitKind,
                           tcx: TyCtxt<'a, 'tcx, 'tcx>,
                           ty: Ty<'tcx>,
                           neg: bool)
-                          -> Result<&'tcx ty::Const<'tcx>, ()> {
+                          -> Result<&'tcx ty::Const<'tcx>, bool> {
```

Comment:
Use an enum instead of a boolean? That way you can communicate the meaning of the different states, not only in the method signature, but also at the return point and when being checked.